internal/tests/cluster/sequential/session_cleanup_test.go

```diff
@@ -163,21 +161,22 @@ func testWorkerSessionCleanupSingle(burdenCase timeoutBurdenType) func(t *testin
 			// Wait on worker, then check controller
 			sess.ExpectConnectionStateOnWorker(ctx, t, w1, session.StatusClosed)
 			sess.ExpectConnectionStateOnController(ctx, t, c1.Controller().ConnectionRepoFn, session.StatusConnected)
-
 		default:
 			// Should be closed on both worker and controller. Wait on
 			// worker then check controller.
 			sess.ExpectConnectionStateOnWorker(ctx, t, w1, session.StatusClosed)
 			sess.ExpectConnectionStateOnController(ctx, t, c1.Controller().ConnectionRepoFn, session.StatusClosed)
 		}
 
+		time.Sleep(15 * time.Second)
```

Comment:
Why is this necessary?

Reply:
Thanks for pointing that out, I actually wanted to bring this issue up. Below this, there is a command `sConn.TestSendRecvFail(t)` that will randomly fail because the error that gets returned is not a net.ErrClosed type, but a connection reset error.

I'm not sure why it's flaky and I believe this behavior is consistent on main. I feel like adding this sleep command has reduced the chances of the test failing, but I could be wrong.

Reply:
I was hoping that someone might have an idea of what we can poll instead of using Sleep.

Reply:
Looks to me like we could actually poll on c.testSendRecv(t), but we'll need a new helper function for that. You could copy TestSendRecvFail and modify it to eventually fail instead of immediately fail. Something like `TestSendRecvEventuallyFail`, maybe take a timeout as a parameter.